app/questionnaire/resolver.py

```diff
@@ -0,0 +1,35 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from decimal import Decimal
+from typing import Iterable, Mapping, TypeAlias
+
+from markupsafe import Markup
+
+from app.data_models.data_stores import DataStores
+from app.questionnaire.questionnaire_schema import QuestionnaireSchema
+from app.utilities.types import LocationType
+
+ResolverTypes = None | str | int | Decimal | list | dict
+ResolveTypes = Markup | list[Markup]
+IntOrDecimal = int | Decimal
+ResolvedAnswerList = list[ResolveTypes | None]
+
+ResolveEscapedTypes: TypeAlias = Markup | list[Markup]
+
+
+@dataclass
+class Resolver(ABC):
```

Comment:
Do we need this abstract class? I don't think this makes sense since there is only one instance of a "Resolver" class? So all of the stuff from ValueSourceResolver would be in here if we did this/nothing would be overridden since they are be the same?

Reply:
It was implemented as a pre measure. I have updated VSR and RuleEvaluator, where RuleEvaluator uses type annotations for the type hinting.